client/lib/plans/features-list.tsx

```diff
@@ -319,7 +322,9 @@ export type FeatureObject = {
 	getCompareSubtitle?: () => TranslateResult;
 	getIcon?: () => string | { icon: string; component: MemoExoticComponent< any > } | JSX.Element;
 	isPlan?: boolean;
+	planDefault?: boolean;
```

Comment:
I understand how `planDefault` is used. I had a slightly different expectation in mind, but I can see how/why it ended up being used this way. What's happening is we have these different feature objects in `FeatureList` that can be used for storage that comes with the plan, and then a series of others that can be used as add-ons (so they cannot be plan defaults).

My main concern is we are adding a storage-specific property to `FeatureObject` that also implies categorization (most of everything else looks quite generic on that structure - like titles, descriptions, slugs, etc.). 

I don't think I have an ideal alternative, but will put a suggestion on the table in case it rings any bells:

1. We can return `StorageOption` from the getter `get2023PricingGridSignupStorageOptions` (so its type becomes `() => StorageOption[]`) and mark things as "default" or "add-on" in there e.g. `() => [ { slug: FEATURE_1GB_STORAGE, planDefault: true } ]` (no need to grab the actual feature object at this point either). 

    - then we extend this type for the grid components to include the `FeatureObject` like needed for the grid implementation i.e. it could just become `PlanGridStorageOption: StorageOption & { feature: FeatureObject }`.

2. I was going to propose that we rename the property `planDefault` (on `FeatureObject`) to `isAddOn`, that way the classification feels a little more direct (it does to me). but then I realize that other plan features may potentially be served as add-ons (in theory, not sure if that's the case).

Reply:
> I don't think I have an ideal alternative, but will put a suggestion on the table in case it rings any bells. Either/or:

Mulling this over...

Reply:
> but then I realize that other plan features may potentially be served as add-ons (in theory, not sure if that's the case).

This is tricky. As far as I could tell when I started this PR, other add-ons in this file were still features . They had "add-on" in the description, but they were describing a functionality inherent to a plan.

https://github.com/Automattic/wp-calypso/blob/1a38b6e6d17fa300f2b12571e9b27ead6de0d33b/client/lib/plans/features-list.tsx#L2075-L2082

https://github.com/Automattic/wp-calypso/blob/1a38b6e6d17fa300f2b12571e9b27ead6de0d33b/client/lib/plans/features-list.tsx#L2345-L2355

Reply:
I'm going the route of renaming the planDefault to isAddOn because it, for the time being seems more intuitive to me. It would also be easy to reverse that decision and switch gears if we realize that it's not the right direction.